test/test_images/runtime/handlers/cgroup.go

```diff
@@ -17,26 +17,43 @@ limitations under the License.
 package handlers
 
 import (
+	"log"
 	"os"
 	"strconv"
 	"strings"
 
 	"knative.dev/serving/test/types"
 )
 
-// cgroupPaths is the set of cgroups probed and returned to the
+// cgroupV1Paths is the set of cgroups probed and returned to the
 // client as Cgroups.
-var cgroupPaths = []string{
+var cgroupV1Paths = []string{
 	"/sys/fs/cgroup/memory/memory.limit_in_bytes",
 	"/sys/fs/cgroup/cpu/cpu.cfs_period_us",
 	"/sys/fs/cgroup/cpu/cpu.cfs_quota_us",
 	"/sys/fs/cgroup/cpu/cpu.shares"}
 
+var cgroupV2Paths = []string{
+	"/sys/fs/cgroup/memory.max",
+	"/sys/fs/cgroup/cpu.max",
+	"/sys/fs/cgroup/cpu.weight"}
+
 var (
 	yes = true
 	no  = false
 )
 
+func cgroupPaths() []string {
+	cgroupv2File := "/sys/fs/cgroup/cgroup.controllers"
+	_, err := os.Stat(cgroupv2File)
```

Comment:
k8s docs https://kubernetes.io/docs/concepts/architecture/cgroups/ mentions `stat -fc %T /sys/fs/cgroup/
` to determine cgroup v2 or not. But verifying of existence of `"/sys/fs/cgroup/cgroup.controllers"` should be alright as well. (it makes much simpler.)

Actually k8s also uses it for test https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20%22%2Fsys%2Ffs%2Fcgroup%2Fcgroup.controllers%22&type=code